templates/al2/provisioners/install-worker.sh

```diff
@@ -392,9 +392,13 @@ if [[ "$CACHE_CONTAINER_IMAGES" == "true" ]] && ! [[ ${ISOLATED_REGIONS} =~ $BIN
   sudo systemctl enable containerd
 
   K8S_MINOR_VERSION=$(echo "${KUBERNETES_VERSION}" | cut -d'.' -f1-2)
+  PREVIOUS_K8S_MINOR_VERSION=$(echo ${KUBERNETES_MINOR_VERSION:0:1}."$((${KUBERNETES_MINOR_VERSION:2:3} - 1))")
```

Comment:
I think we want to just skip image caching in this case; this will only happen for a brief period early in the version support cycle, before the version is GA on EKS

Reply:
Okay cool! So then the correct behavior should be:
If cached results is nil, then we skip everything up to the [SSM agent](https://github.com/awslabs/amazon-eks-ami/blob/main/templates/al2/provisioners/install-worker.sh#L469)

Reply:
Yep! Instead of wrapping that whole block in an `if`, you could do something like move it into a `function` and then return-early if you find there's nothing to cache